packages/core/utils/src/config.js

```diff
@@ -36,9 +37,10 @@ export async function resolveConfig(
 
   for (const filename of filenames) {
     let file = path.join(filepath, filename);
-    let exists = existsCache.has(file)
-      ? existsCache.get(file)
-      : await fs.exists(file);
+    let exists =
+      existsCache.has(file) && (!opts || !opts.noCache)
```

Comment:
can this grow indefinitely? can't it be invalidated by files being removed?

Reply:
`existsCache.has(file) && !opts?.noCache` maybe?

Reply:
Ideally anything that needs to be invalidated would be invalidated in the same way at the same time, which at this point is the `RequestIGraph`. I'm not really sure we should keep the `existsCache` at all. The only other things using it right now are `loadEnv` which only happens once (right?) and `installPackage`, which probably doesn't happen often enough to notice a performance improvement.